lib/providers/combined_repository_status.dart

```diff
@@ -55,22 +43,58 @@ Future<StoredCombinedRepositoryStatus> combinedRepositoryStatus(
     return StoredCombinedRepositoryStatus();
   }
 
+  final id = await ref.watch(
+      _fetchAndStoreCombinedRepositoryStatusProvider(repo, pullRequest).future);
+  return box.get(id)!;
+}
+
+Future<void> combinedRepositoryStatusRefresh(
+    WidgetRef ref, StoredRepository repo, StoredPullRequest pullRequest) async {
+  ref.invalidate(
+      _fetchAndStoreCombinedRepositoryStatusProvider(repo, pullRequest));
+  await ref.watch(
+      _fetchAndStoreCombinedRepositoryStatusProvider(repo, pullRequest).future);
+}
+
+@riverpod
+Future<int> _fetchAndStoreCombinedRepositoryStatus(
+    Ref ref, StoredRepository repo, StoredPullRequest pullRequest) async {
+  final slug = repo.data.slug();
+  final commitSha = pullRequest.data.head.sha;
+
+  final store = await ref.watch(objectBoxProvider.future);
+  final box = store.box<StoredCombinedRepositoryStatus>();
+
+  // Fetch the combined status from GitHub.
   final status = await ref
-      .watch(_githubCombinedRepositoryStatusProvider(slug, commitSha).future);
+      .watch(_githubCombinedRepositoryStatusProvider(repo, pullRequest).future);
+
+  // Delete existing combined status for the same commit.
+  box
+      .query(StoredCombinedRepositoryStatus_.repo.equals(slug.fullName) &
+          StoredCombinedRepositoryStatus_.sha.equals(commitSha))
```

Comment:
Is this a bitwise-and or some kind of magic?

Reply:
It's ".and", no magic.

Reply:
I mean, they have both `&` and `&&` as bitwise and logical respectively in Dart, so it's pretty odd that `&` is used for logical "and" here. :(

https://dart.dev/language/operators#logical-operators